src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -1631,6 +1632,65 @@ case class BinaryOp(posn: Position, lhs: AST, operation: String, rhs: AST) exten
       s.r.findFirstIn(t).isDefined
     }
 
+    case ("+" | "*" | "-" | "/", TArray(elementType)) =>
+      val f: (Any, Any) => Any = (operation, elementType) match {
```

Comment:
I sent you a PR that simplifies this a tad more

https://github.com/tpoterba/hail/pull/5